src/OrchardCore.Modules/Orchard.Autoroute/Drivers/AutoroutePartDisplay.cs

```diff
@@ -61,15 +73,36 @@ public override IDisplayResult Edit(AutoroutePart autoroutePart)
         public override async Task<IDisplayResult> UpdateAsync(AutoroutePart model, IUpdateModel updater)
         {
             await updater.TryUpdateModelAsync(model, Prefix, t => t.Path);
-            
+
             var httpContext = _httpContextAccessor.HttpContext;
-            
             if (httpContext != null && await _authorizationService.AuthorizeAsync(httpContext.User, Permissions.SetHomepage))
             {
                 await updater.TryUpdateModelAsync(model, Prefix, t => t.SetHomepage);
             }
 
+            await ValidateAsync(model, (key, message) => updater.ModelState.AddModelError(Prefix, key, message));
+
             return Edit(model);
         }
+
+        private async Task ValidateAsync(AutoroutePart autoroute, Action<string, string> reportError)
+        {
+            var invalidCharacters = ":?#[]@!$&'()*+,.;=<>\\|% ".ToCharArray();
```

Comment:
You are paying the price of our mistakes in O1 ;) I looked at the code and now I can see the many things you ported. Like the message names, Regex, ... I wonder how we chose this list. Maybe from the IIS default set of excluded chars... That's why we need code comments :/